include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	void setInputAttribValue(unsigned int controlArrayLocation, float floatValue, bool boolValue);
+	// calculates the ideal text color
+	QColor getTextColorFromBaseColor(QColor baseColor);
+	// calculates a replacement background fill color
+	QColor getFillColorFromBaseColor(QColor baseColor);
+	// cuts the string to displayedLength(in px) size (estimated)
+	QString getTextFromDisplayLength(QString text, unsigned int displayLength);
+	// context menu actions
+	void addDefaultActions(QMenu* menu, QString controlDisplayText);
+
+	// inputDialog
+	std::pair<float, float> showCoordInputDialog();
+	float showInputDialog(float curInputValue);
+
+	// selection
+	// searches VectorGraphDataArray-s to select
+	// near clicked location
+	void selectData(int mouseX, int mouseY);
+	// searches for point in a given VectorGraphDataArray
+	// returns found location, when a point
+	// was found in the given distance
+	// else it returns -1
+	int searchForData(int mouseX, int mouseY, float maxDistance, VectorGraphDataArray* dataArray, bool isCurved);
+
+	// if the mouse is not moved
+	bool m_mousePress;
+	// decides addition or deletion
+	bool m_addition;
+
+	// radius, rx = ry
+	unsigned int m_pointSize;
+	unsigned int m_fontSize;
+	// draw simplified lines
+	bool m_isSimplified;
+	// true when applyDefaultColors is called, used when defaultColors are loading
+	bool m_isDefaultColorsApplyed;
+	QPixmap m_background;
+	// for 1 draw, it will use the VectorGraphDataArray
+	// m_bakedSamples without calling getSamples()
+	bool m_useGetLastSamples;
+
+	// if m_isLastSelectedArray == true then
+	// m_selectedArray can be used
+	// else if m_isSelected == false then
+	// m_selectedLocation amd m_selectedArray should not be used
+	unsigned int m_selectedLocation;
+	unsigned int m_selectedArray;
+	bool m_isSelected;
+	bool m_isCurveSelected;
+	// if m_selectedArray was the last array selected
+	bool m_isLastSelectedArray;
+
+	unsigned int m_graphHeight;
+	unsigned int m_controlHeight;
+	// displayed control count (+1 because of the ">>" button in editing mode)
+	unsigned int m_controlDisplayCount;
+	unsigned int m_controlDisplayPage;
+	bool m_isEditingActive;
+	std::vector<QString> m_controlText;
+	std::vector<QString> m_controlLineEffectText;
+	std::vector<bool> m_controlIsFloat;
+
+	std::pair<int, int> m_lastTrackPoint;
+	std::pair<int, int> m_lastScndTrackPoint;
+
+	
+	// default VectorGraphDataArray colors
+	// applyed in constructor
+	QColor m_vectorGraphDefaultAutomatedColor;
+
+	QColor m_vectorGraphDefaultLineColor;
+	QColor m_vectorGraphDefaultActiveColor;
+	QColor m_vectorGraphDefaultFillColor;
+	QColor m_vectorGraphSecondaryLineColor;
+	QColor m_vectorGraphSecondaryActiveColor;
+	QColor m_vectorGraphSecondaryFillColor;
+};
+
+} // namespace gui
+
+class LMMS_EXPORT VectorGraphModel : public Model, public JournallingObject
+{
+Q_OBJECT
+public:
+	VectorGraphModel(unsigned int arrayMaxLength, Model* parentIn, bool defaultConstructed);
```

Comment:
I would like to keep `parentIn`. `parent` sounds too general to me and since this class inherits Q_Object, where `parent` could be a variable `parentIn` looks less confusing.

Reply:
`parent` is the standard name for Qt objects. Use that please.

Reply:
All right, thanks for the answer.